### ISSUE TITLE
Problem with WCF Asynchronous Service - Thread sync issue without pre-compilation bug

### ISSUE THREAD

**Issue by Mariana1980** [expressmapper/ExpressMapper#61]
Hi Yuriy!!!!
Now, we are testing a wcf service with ExpressMapper and we found a problem.
The first time , the mapper throws a  null pointer exception. But, the second time runs correctly in the same instance.
We have an example [here:] (https://www.dropbox.com/s/hghq7hxl64bhuwh/Ejemplo.zip?dl=0)

Thanks a lot!.

**Comment by anisimovyuriy** [comment]
Hi Mariana1980,

Unfortunately I couldn't reproduce the error - it worked for me.
Also I tried the following:

```
public class Source{
    public int Id {get;set;}
}

public class Target{
    public int Id {get;set;}
}

var src = new Source
    {
         Id = 5
    };
var target = Mapper.Map<Source, Target>(src);
var newSrc = Mapper.Map<Target, Source>(target);
```

I haven't received any exceptions and mappings were correct.
Could you please maybe give me some more details how to reproduce the issue?

Thanks,
Yuriy

**Comment by Mariana1980** [comment]
Hi Yuriy!
There is a word document with the step for reproduce the error [here] (https://www.dropbox.com/s/nk6yjvggl4mejwa/Error%20WS.docx?dl=0). 

And the latest version of application with try-catch [here] (https://www.dropbox.com/s/x4f9sxe1dx9rsai/WebService.zip?dl=0)

Thanks!

**Comment by juniorkrvl** [comment]
Hello guys, I had the same issue with my WCF. The first time I run, I get this stack trace and then it works without the error.

In my case, the WCF is self hosted, so I register my maps when Windows Service is started.

```
System.NullReferenceException: Object reference not set to an instance of an object.
   em ExpressMapper.TypeMapperBase`2.MapTo(T src, TN dest) na C:\SourceCode\ExpressMapper\ExpressMapper NET40\TypeMapperBase.cs:line 243
   em ExpressMapper.MappingServiceProvider.MapInternal[T,TN](T src, TN dest, Boolean dynamicTrial) na C:\SourceCode\ExpressMapper\ExpressMapper NET40\MappingServiceProvider.cs:line 273
   em ExpressMapper.MappingServiceProvider.Map[T,TN](T src, TN dest) na C:\SourceCode\ExpressMapper\ExpressMapper NET40\MappingServiceProvider.cs:line 246
   em ExpressMapper.Mapper.Map[T,TN](T src, TN dest) na C:\SourceCode\ExpressMapper\ExpressMapper NET40\Mapper.cs:line 43
```

**Comment by anisimovyuriy** [comment]
Hey Guys,

Sorry for not responding long time - I was kind of in a big rush, but since now I'll be more dedicated.
Let me try to reproduce Mariana's source code and instructions and I'll get back to you very soon :)

Thanks,
Yuriy

**Comment by anisimovyuriy** [comment]
Thanks a lot guys - got it. It's a thread sync bug using dynamic mapping (Trying to map without registration). I'm going to fix it mean while please in your AppInitialize or you can use Factory - use Mapper.Compile() after mapping registration.

Thanks for your patience!

**Comment by anisimovyuriy** [comment]
Fixed and will be released with Expressmapper 1.8

**Comment by juniorkrvl** [comment]
Yuri, thank you for your response. I did the Mapper.Compile() and in my case it worked very well!
